contracts/base/Dispatcher.sol

```diff
@@ -132,37 +140,45 @@ abstract contract Dispatcher is NFTImmutables, Payments, V2SwapRouter, V3SwapRou
                     // 0x08 <= command < 0x10
                 } else {
                     if (command == Commands.V2_SWAP_EXACT_IN) {
-                        // equivalent: abi.decode(inputs, (address, uint256, uint256, bytes, bool))
+                        // equivalent: abi.decode(inputs, (address, uint256, uint256, bytes, bool, bytes32, address))
                         address recipient;
                         uint256 amountIn;
                         uint256 amountOutMin;
                         bool payerIsUser;
+                        bytes32 v2PairInitCodeHash;
+                        address v2Factory;
```

Comment:
To keep the contract interface simpler, I prefer to save these code hashes and factory addresses in contract code (as immutables), and define another input parameter called `v2Folk`, which is basically an enum, to map to corresponding immutables.

It also helps to reduce calldata size, and simplify the SDK code.

Reply:
If there is no gas consumption issue, I prefer this solution since we don’t have to support forks one by one

Reply:
Understand your point. I guess gas is a less important concern here, but the interface complexity matters more for me: caller should not be responsible to tell us the code hashes and factory addresses.

We do need to revise the contract and re-deploy to support more forks, but I don't expect that to happen too often - there won't be many forks worth to support.

Reply:
Actually I forgot another point: even with the current solution, we still need to add callback functions for each fork - anyway a new fork won't be supported without contract code changes.